common/views/components/SearchLogger/SearchLogger.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+import { useEffect, useState } from 'react';
+import Router from 'next/router';
+
+const segmentId = '78Czn5jNSaMSVrBq2J9K4yJjWxh6fyRI';
+
+function routeChangeComplete(analytics) {
+  return url => {
+    if (
+      window.location.pathname.match('/works') &&
+      'URLSearchParams' in window
+    ) {
+      const workPageMatch = window.location.pathname.match(
+        /^\/works\/([a-z0-9]+)$/
+      );
+      const itemPageMatch = window.location.pathname.match(
+        /^\/works\/([a-z0-9]+)\/items$/
+      );
+      const query = Array.from(
+        new URLSearchParams(window.location.search)
+      ).reduce(function(acc, keyVal) {
+        acc[keyVal[0]] = keyVal[1];
+        return acc;
+      }, {});
+      const toggles = document.cookie.split(';').reduce(function(acc, cookie) {
```

Comment:
This still feels like the nastiest bit - @wellcometrust/experience-devs any ideas?
Maybe I can pull this out into a `CataloguePageLayout` which takes an analytics `event`?